cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -321,19 +323,31 @@ protected List<ContainerPort> getContainerPortList() {
     }
 
     protected List<Volume> getVolumes(boolean isOpenShift) {
-        return List.of(VolumeUtils.createTempDirVolume(templatePod),
-                createSecretVolume(TLS_CC_CERTS_VOLUME_NAME, CruiseControlResources.secretName(cluster), isOpenShift),
-                createSecretVolume(TLS_CA_CERTS_VOLUME_NAME, AbstractModel.clusterCaCertSecretName(cluster), isOpenShift),
-                createSecretVolume(API_AUTH_CONFIG_VOLUME_NAME, CruiseControlResources.apiSecretName(cluster), isOpenShift),
-                createConfigMapVolume(CONFIG_VOLUME_NAME, CruiseControlResources.configMapName(cluster)));
+        List<Volume> volumes = new ArrayList<>();
+        volumes.add(VolumeUtils.createTempDirVolume(templatePod));
+        volumes.add(createSecretVolume(TLS_CC_CERTS_VOLUME_NAME, CruiseControlResources.secretName(cluster), isOpenShift));
+        volumes.add(createSecretVolume(TLS_CA_CERTS_VOLUME_NAME, AbstractModel.clusterCaCertSecretName(cluster), isOpenShift));
+        volumes.add(createSecretVolume(API_AUTH_CONFIG_VOLUME_NAME, CruiseControlResources.apiSecretName(cluster), isOpenShift));
+        volumes.add(createConfigMapVolume(CONFIG_VOLUME_NAME, CruiseControlResources.configMapName(cluster)));
+        
+        if (templatePod != null) {
+            addAdditionalVolumes(templatePod, volumes);
+        }
```

Comment:
Would it make sense to do the null checks inside the `addAdditionalVolumes` method? It would make the code flow more nicely here and you anyway have inside the method the check that `templatePod.getVolumes != null`. So I think it would make more sense to have the checks together and in one place.